[CI:DOCS] Add socket_activation.md

eriksjolund opened the pull request:
Closes #14908

Signed-off-by: Erik Sjölund <erik.sjolund@gmail.com>

<!--
Thanks for sending a pull request!

Please make sure you've read our contributing guidelines and how to submit a pull request (https://github.com/containers/podman/blob/main/CONTRIBUTING.md#submitting-pull-requests).

In case you're only changing docs, make sure to prefix the pull-request title with "[CI:DOCS]". That will prevent functional tests from running and save time and energy.

Finally, be sure to sign commits with your real name. Since by opening
a PR you already have commits, you can add signatures if needed with
something like `git commit -s --amend`.
-->

#### Does this PR introduce a user-facing change?

<!--
If no, just write `None` in the release-note block below. If yes, a release note
is required: Enter your extended release note in the block below. If the PR
requires additional action from users switching to the new release, include the
string "action required".

For more information on release notes, please follow the Kubernetes model:
https://git.k8s.io/community/contributors/guide/release-notes.md
-->

```release-note
None
```

vrothberg reviewed (commented):
Nice work, @eriksjolund!

@rhatdan @giuseppe @TomSweeneyRedHat PTAL

eriksjolund commented:
A side-note:

@rhatdan and I was [discussing](https://github.com/containers/podman/discussions/13390#discussioncomment-2858188) the possibility create a related blog post for
https://www.redhat.com/sysadmin/topics/containers

Here is my initial attempt:
https://github.com/eriksjolund/podman-socket-activation/blob/rework_text/blog.md
(For me it would be fine to be a co-author too if you want to modify it)

I also wrote another text
https://github.com/eriksjolund/podman-socket-activation/blob/rework_text/RestrictAddressFamilies.md
(that might be too avant garde for a redhat blog or the podman web page)

TomSweeneyRedHat commented on docs/tutorials/podman-socket-activation.md:
```diff
@@ -0,0 +1,208 @@
+
+### Note about SELinux
+
+> **Note**
```
I'm not sure I like how the "Note" resolves.  I think I'd drop it.

eriksjolund replied:
I removed both notes. The importance of the first note will eventually fade away as people are using newer versions of container-selinux.

The second note is probably more of interest to the end users that are super interested and not to the general end user. 

```
### Note about SELinux

> **Note**
> If your computer is running SELinux, you need to have [__container-selinux 2.186.0__](https://github.com/containers/container-selinux)
> or newer installed. If container socket activation via Podman does not work and you are using an older version of
> container-selinux, add `--security-opt label=disable` to `podman run` as a work around.

### Note about running systemd in container

> **Note** There is currently no way to pass a socket-activated socket into a container when the container is running
> systemd. In other words, the executable __/usr/lib/systemd/systemd__ does not support socket activation.
> (There is a feature request related to it https://github.com/systemd/systemd/issues/17764)
```

TomSweeneyRedHat commented:
@eriksjolund you've been rocking the tutorials of late.  Can you also please update the README.md with links to the new tutorials and a small description of each?

eriksjolund commented:
> @eriksjolund you've been rocking the tutorials of late. Can you also please update the README.md with links to the new tutorials and a small description of each?

I added a link to the socket activation tutorial. (I'll wait with adding a link to the performance tutorial because I don't know how long it will take to finish it)


Another thing. I modified this PR to build a local HTTPD container instead of  of using _ghcr.io/eriksjolund/socket-activate-httpd_. The reason is that I wanted to avoid the responsibility to keep that container image up to date to the latest security fixes in HTTPD.

eriksjolund commented:
First thought I should write something about the need to use the Linux operating system to be able to use socket activation.
I had difficulties coming up with a text about that  so I just skipped that part.
Maybe it's possible to use Windows or Mac with a Linux VM that use VSOCK?
I'm a little confused about the topic so I skipped writing about that too.

I remove the draft status.

rhatdan commented:
/approve
LGTM
@eriksjolund Great work.  Did you ever publish the full blogs?
We can wait for @vrothberg to merge.

vrothberg reviewed (commented):
/lgtm

eriksjolund commented:
> @eriksjolund Great work. Did you ever publish the full blogs?

@rhatdan Yesterday I split out the tutorial about using the systemd directive [__RestrictAddressFamilies__](https://www.freedesktop.org/software/systemd/man/systemd.exec.html#RestrictAddressFamilies=)
into the new GitHub repository https://github.com/eriksjolund/podman-RestrictAddressFamilies
On Monday or Tuesday I plan to write to tweet about it and also about the new [Podman socket activation tutorial](https://github.com/containers/podman/blob/main/docs/tutorials/socket_activation.md)

I rewrote the [potential RedHat blog](https://github.com/eriksjolund/podman-socket-activation/blob/rework_text/blog.md) a bit. (I removed the SELinux part and instead discussed privileges in more general terms).
Do you think the blog could be published under https://www.redhat.com/sysadmin/topics/containers ?
If so, how should I proceed?
